### PR TITLE
Rich text for copy-on-click for permalinks

### PR DESCRIPTION
**Pull request by jjrsylvestre**
Can be simplified a lot if Rob adds a data-description attribute to things that should be assigned a permalink.

### CODE REVIEW

**Comment by jjrsylvestre** [comment]
I just did a raw `await` in the `copyToClipboard` function in `src/features/global/globalSlice.ts` --- I don't know if you need this to be a "Thunk" instead.

File: src/features/global/globalSlice.ts [hunk 1]
```diff
@@ -41,9 +46,43 @@ const globalThunks = {
...
         "global/copyToClipboard",
-        async (text: string, { dispatch, getState }) => {
-            copy(text);
-            toastr.success("Copied to clipboard", text);
+        async ( linkDetails : PermalinkDetails, { dispatch, getState }) => {
+            if (!navigator.clipboard) {
+                return;
+            }
+            let linkText = linkDetails.title ?? linkDetails.url;
+            let link = "<a href=\"" + linkDetails.url + "\">" + linkText + "</a>";
```
[the remark this answers is not in the document]
Comment (jjrsylvestre) [reply src/features/global/globalSlice.ts:54]: Yeah, I probably wrote that line before I remembered that such things existed

File: src/features/global/globalSlice.ts [hunk 2]
```diff
@@ -41,9 +46,43 @@ const globalThunks = {
...
+            let text_fallback;
+            if (linkText != linkDetails.url) {
+                text_fallback = linkDetails.title + " \r\n" + linkDetails.url;
```
[the remark this answers is not in the document]
Comment (jjrsylvestre) [reply src/features/global/globalSlice.ts:57]: David F inserted the `\r` when I wrote the original copy-permalink-to-clipboard code in the old javascript. My understanding is that Windows requires combo `\r\n`, but I don't have a Windows machine to check right now.

**Comment by siefkenj** [comment]
Thanks for the hard work!

**Pull request merged**